std/math.d

```diff
@@ -7607,3 +7607,132 @@ T truncPow2(T)(const T val) if (isFloatingPoint!T)
         assert(truncPow2(T.init).isNaN);
     }
 }
+
+/**
+Check whether a number is an integer power of two.
+
+Note that only positive numbers can be integer powers of two. This
+function always return `false` if `x` is negative or zero.
+
+Params:
+    x = the number to test
+
+Returns:
+    `true` if `x` is an integer power of two.
+*/
+bool isPowerOf2(X)(const X x) pure @safe nothrow @nogc
+    if (isNumeric!X)
+{
+    static if (isFloatingPoint!X)
+    {
+        if (__ctfe)
+        {
+            // TODO: Simplify this as more floating-point functions are made to work at CTFE
+            const isFinite = -real.infinity < x && x < real.infinity;
+            if (!isFinite)
+                return false;
+
+            const logb = {
+                auto ret = log(x) / log(2.0L);
+                ret += ret > 0? 0.5L : -0.5L;
+                return cast(long)ret;
+            }();
+            const exp2 = pow(2.0L, logb);
+            return (x == exp2) && (x != 0);
+        }
+        else
+        {
+            int exp;
+            const X sig = frexp(x, exp);
+
+            return (exp != int.min) && (sig is cast(X)0.5L);
+        }
+    }
+    else
+    {
+        static if (isSigned!X)
+            alias W = typeof(x + 0);
+        else
+            alias W = typeof(x + 0u); // For ubyte and ushort W should be uint
+
+        static if (is(X == W))
+        {
+            static if (isSigned!X)
+                return ((x & -x) == x) && (x > -x);
+            else
+                return (x & -x) > (x - 1); // Credit: Matthias Bentrup
```

Comment:
Who is Matthias Bentrup and why is he credited for a commo knowledge algorithms? :-)